### PR TITLE
RFC for structured technical decision-making process

### PR DESCRIPTION
**Pull request by snizhana-dynnyk**
towards https://github.com/giantswarm/giantswarm/issues/30631

This RFC extends [RFC and decision-making process](https://handbook.giantswarm.io/docs/rfcs/decision-process/) with an RFC structure and introduces metadata necessary for submitting a decision proposal.

### CODE REVIEW

File: technical-decision-making-process/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+creation_date: 2024-05-22
+issues:
+- https://github.com/giantswarm/giantswarm/issues/30631
+owners:
+- https://github.com/orgs/giantswarm/teams/team-horizon
+state: review
+summary: This contains the process for proposing, discussing, and formalizing technical decisions. It also introduces the RFC structure.
+---
+
+# Structured way to propose, discuss, and formalize technical decisions within an organization
```
Comment (JosephSalisbury) [inline technical-decision-making-process/README.md:11]:
```suggestion
# Structured way to propose, discuss, and formalize technical decisions within an organization

```

nit, would be nice to have a newline here

File: technical-decision-making-process/README.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+### Problem statement
+
+The current technical decision-making process within our organization is inconsistent and inefficient, particularly when decisions impact multiple teams. This lack of a well-defined process leads to confusion, delays, and suboptimal decisions.
+We need a structured approach to proposing, discussing, and formalizing decisions to improve collaboration and decision quality.
+
+The first [RFC and decision making process](https://handbook.giantswarm.io/docs/rfcs/decision-process/) proposed a file structure and introduced an overview of RFCs in the [Handbook](https://handbook.giantswarm.io/docs/rfcs/).
+It also suggested a process for submitting an RFC for a technical decision.
+
+The current RFC extends the latter with RFC structure and introduces fields necessary for submitting a decision proposal.
```
Comment (JosephSalisbury) [inline technical-decision-making-process/README.md:20]:
```suggestion
This RFC extends the latter with a structure and introduces fields necessary for submitting a decision proposal.
```

File: technical-decision-making-process/README.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+summary: This contains the process for proposing, discussing, and formalizing technical decisions. It also introduces the RFC structure.
+---
+
+# Structured way to propose, discuss, and formalize technical decisions within an organization
+### Problem statement
+
+The current technical decision-making process within our organization is inconsistent and inefficient, particularly when decisions impact multiple teams. This lack of a well-defined process leads to confusion, delays, and suboptimal decisions.
+We need a structured approach to proposing, discussing, and formalizing decisions to improve collaboration and decision quality.
+
+The first [RFC and decision making process](https://handbook.giantswarm.io/docs/rfcs/decision-process/) proposed a file structure and introduced an overview of RFCs in the [Handbook](https://handbook.giantswarm.io/docs/rfcs/).
```
Comment (JosephSalisbury) [inline technical-decision-making-process/README.md:17]:
```suggestion
The first [RFC and decision making process](https://github.com/giantswarm/rfc/pull/78/files/) proposed a file structure and introduced an overview of RFCs in the [Handbook](https://handbook.giantswarm.io/docs/rfcs/).
```

File: technical-decision-making-process/README.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+It also suggested a process for submitting an RFC for a technical decision.
+
+The current RFC extends the latter with RFC structure and introduces fields necessary for submitting a decision proposal.
+This enables a more structural approach to making decisions and specifies who the decision maker is, for whom it is mandatory to provide feedback, and in which timeframe. Additionally, it includes a communication plan to ensure clarity and alignment.
```
Comment (JosephSalisbury) [inline technical-decision-making-process/README.md:21]:
```suggestion
This enables a more structural approach to making decisions and specifies who the decision maker is, for whom it is mandatory to provide feedback, and in which timeframe. Additionally, it includes a communication plan to ensure clarity and alignment.

To help with an example, this RFC is itself in the proposed structure
```

File: technical-decision-making-process/README.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+### Deadline
+13.06.2024
+
+### Who is affected / stakeholders
+- Product Owners
+- Platform Architects
+- Team Horizon
+
+### Preferred solution
```
Comment (JosephSalisbury) [inline technical-decision-making-process/README.md:34]:
```suggestion
### Preferred solution

```

also newline nits

File: technical-decision-making-process/README.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+#### 1. Technical decision-making process:
+- Proposal Creation: Any team member can create an RFC document to propose a decision. See [RFC and decision making process](https://handbook.giantswarm.io/docs/rfcs/decision-process/) for a description of submitting an RFC process.
```
Comment (JosephSalisbury) [inline technical-decision-making-process/README.md:36]:
```suggestion
- Proposal Creation: Any team member can create an RFC document to propose a decision. See [RFC and decision making process](https://handbook.giantswarm.io/docs/rfcs/decision-process/) for a description of submitting an RFC.
```

File: technical-decision-making-process/README.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+- Review and Discussion: The RFC is shared with relevant stakeholders for review and feedback.
+- Decision Meeting: If there is no agreement or insufficient feedback before the deadline, the decision owner schedules a formal meeting with relevant stakeholders to discuss the RFC and make a decision.
+- Approval and Implementation: Once agreed upon, the decision is documented, and an implementation plan is created.
+
+#### 2. Template for technical decision RFC:
```
Comment (JosephSalisbury) [inline technical-decision-making-process/README.md:41]: What does '1' and '2' mean in these sections?

File: technical-decision-making-process/README.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+- Adopt a free-form document process:
+  - Pros: Flexibility in document format.
+  - Cons: Lack of structure can lead to confusion and missed details.
+
+### Implementation Plan
+Preparation Phase:
+ - By 13.06.2024: Develop the RFC template and guidelines.
+ - By 1.07.2024: Onboard main stakeholders on the new process.
+
+Pilot Phase:
```
Comment (JosephSalisbury) [inline technical-decision-making-process/README.md:66]: I actually wouldn't bother with this - if we get enough people to agree on the new structure, we can just do it

**Review by JosephSalisbury — COMMENTED** [review]
first pass, i like this

File: technical-decision-making-process/README.md (hunk 7 above)
Comment (snizhana-dynnyk) [reply technical-decision-making-process/README.md:45]: nothing :) just subsections, will remove numbers

File: technical-decision-making-process/README.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+13.06.2024
+
+### Who is affected / stakeholders
+- Product Owners
+- Platform Architects
+- Team Horizon
+
+### Preferred solution
+
+#### 1. Technical decision-making process:
```
Comment (snizhana-dynnyk) [inline technical-decision-making-process/README.md:39]:
```suggestion
#### Technical decision-making process:
```

File: technical-decision-making-process/README.md [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+- Proposal Creation: Any team member can create an RFC document to propose a decision. See [RFC and decision making process](https://handbook.giantswarm.io/docs/rfcs/decision-process/) for a description of submitting an RFC.
+- Review and Discussion: The RFC is shared with relevant stakeholders for review and feedback.
+- Decision Meeting: If there is no agreement or insufficient feedback before the deadline, the decision owner schedules a formal meeting with relevant stakeholders to discuss the RFC and make a decision.
+- Approval and Implementation: Once agreed upon, the decision is documented, and an implementation plan is created.
+
+#### 2. Template for technical decision RFC:
```
Comment (snizhana-dynnyk) [inline technical-decision-making-process/README.md:45]:
```suggestion
####  Template for technical decision RFC:
```

File: technical-decision-making-process/README.md [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+- Adopt a free-form document process:
+  - Pros: Flexibility in document format.
+  - Cons: Lack of structure can lead to confusion and missed details.
+
+### Implementation Plan
+Preparation Phase:
+ - By 13.06.2024: Develop the RFC template and guidelines.
+ - By 1.07.2024: Onboard main stakeholders on the new process.
+
+Pilot Phase:
```
Comment (snizhana-dynnyk) [inline technical-decision-making-process/README.md:70]:
```suggestion
```

File: technical-decision-making-process/README.md [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+ - From 01.07.2024: Pilot the process with one team and gather feedback.
```
Comment (snizhana-dynnyk) [inline technical-decision-making-process/README.md:71]:
```suggestion
```

File: technical-decision-making-process/README.md [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+
```
Comment (snizhana-dynnyk) [inline technical-decision-making-process/README.md:72]:
```suggestion
```

File: technical-decision-making-process/README.md (hunk 8 above)
Comment (snizhana-dynnyk) [reply technical-decision-making-process/README.md]: removed

File: technical-decision-making-process/README.md [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+### Problem statement
+
+The current technical decision-making process within our organization is inconsistent and inefficient, particularly when decisions impact multiple teams. This lack of a well-defined process leads to confusion, delays, and suboptimal decisions.
+We need a structured approach to proposing, discussing, and formalizing decisions to improve collaboration and decision quality.
+
+The first [RFC and decision making process](https://github.com/giantswarm/rfc/pull/78/files/) proposed a file structure and introduced an overview of RFCs in the [Handbook](https://handbook.giantswarm.io/docs/rfcs/).
+It also suggested a process for submitting an RFC for a technical decision.
+
+This RFC extends the latter with a structure and introduces fields necessary for submitting a decision proposal.
+This enables a more structural approach to making decisions and specifies who the decision maker is, for whom it is mandatory to provide feedback, and in which timeframe. Additionally, it includes a communication plan to ensure clarity and alignment.
```
Comment (puja108) [inline technical-decision-making-process/README.md:22]:
```suggestion
This enables a more structured approach to making decisions and specifies who the decision maker is, for whom it is mandatory to provide feedback, and in which timeframe. Additionally, it includes a communication plan to ensure clarity and alignment.
```

File: technical-decision-making-process/README.md [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+  - Pros: No change required.
+  - Cons: Inefficiencies and inconsistencies remain.
+
+- Adopt a free-form document process:
+  - Pros: Flexibility in document format.
+  - Cons: Lack of structure can lead to confusion and missed details.
+
+### Implementation Plan
+Preparation Phase:
+ - By 13.06.2024: Develop the RFC template and guidelines.
```
Comment (puja108) [inline technical-decision-making-process/README.md:67]: I'd suggest also adapting https://handbook.giantswarm.io/docs/rfcs/decision-process/ to reflect the process/structure proposed here.

Most probably there, it would also help to give some guidline as to when to create RFCs (more than one team affected, bigger change within a team that should get feedback from security and sig arch,..) and who usually to add as stakeholders if in doubt (sig arch, sig security, ...). cc @snizhana-dynnyk as you suggested I suggest that in my review here, but it feels it fits more in the format of the other RFC that anyway needs an update.

**Review by puja108 — APPROVED** [review]
LGTM

**Review by weatherhog — APPROVED** [review]

**Review by anvddriesch — APPROVED** [review]

File: technical-decision-making-process/README.md [hunk 16]
```diff
@@ -0,0 +1,84 @@
...
+---
+
+# Structured way to propose, discuss, and formalize technical decisions within an organization
+
+### Problem statement
+
+The current technical decision-making process within our organization is inconsistent and inefficient, particularly when decisions impact multiple teams. This lack of a well-defined process leads to confusion, delays, and suboptimal decisions.
+We need a structured approach to proposing, discussing, and formalizing decisions to improve collaboration and decision quality.
+
+The first [RFC and decision making process](https://github.com/giantswarm/rfc/pull/78/files/) proposed a file structure and introduced an overview of RFCs in the [Handbook](https://handbook.giantswarm.io/docs/rfcs/).
```
Comment (marians) [inline technical-decision-making-process/README.md:18]: The RFC added in PR https://github.com/giantswarm/rfc/pull/78/ is now published as https://handbook.giantswarm.io/docs/rfcs/decision-process/. I would prefer using the final URL instead of the PR one.

**Comment by JosephSalisbury** [comment]
oh yeah, ping @AndiDog, i assume you have thoughts here <3

**Comment by stone-z** [comment]
This looks fine as a structural starting point.

Anticipatory suggestions:
- it might need to be more clear when review is required from an individual, since there can be a strong bystander effect with pinging a whole SIG
- there needs to be enough time allowed to review, considering individual AFKs, holidays, etc. One month seems like a reasonable ballpark, but I can imagine some folks might want to move faster than that. That expectation should probably be documented to keep everyone on the same page.

**Review by AndiDog — COMMENTED** [review]
I'm not convinced that a template will solve things. What about we instead do a test run of the proposed format (side note: YYYY-MM-DD date format everywhere please)? Like, trying this on an actual, current technical decision.

What I'm missing here is that having a template by itself doesn't trigger people to put decisions into writing. Also, putting an artificial deadline in there has never worked for any RFC-like document, which is why it was particularly _not_ suggested for RFC fields.

tl;dr: veto that this won't work

**Comment by pipo02mix** [comment]
IMO, one of the reasons I see ppl making decisions within the team and not relying on RFC is the unfamiliarity with the process and the ambiguity of who should do it. My proposal is the architect person(s) in the team is the one that needs to drive it, we can train these roles and run some dryruns

File: technical-decision-making-process/README.md (hunk 16 above)
Comment (puja108) [reply technical-decision-making-process/README.md:18]: But then at least link the repo URL instead of the PR, right?
like https://github.com/giantswarm/rfc/blob/main/decision-process/README.md or are we worried we might change the process and want to make sure we link the original PR only?

Comment (Rotfuks) [reply technical-decision-making-process/README.md:18]: When we reference the rfc here we reference it to be read, or not? So if the best touchpoint to read it is the handbook, why not link that? 
But yes, at least if we want the repo to be the main touchpoint, we should use the repo url instead of the PR to not have to update this one whenever there is a change :)

File: technical-decision-making-process/README.md [hunk 17]
```diff
@@ -0,0 +1,84 @@
...
+  - Cons: Inefficiencies and inconsistencies remain.
+
+- Adopt a free-form document process:
+  - Pros: Flexibility in document format.
+  - Cons: Lack of structure can lead to confusion and missed details.
+
+### Implementation Plan
+Preparation Phase:
+ - By 13.06.2024: Develop the RFC template and guidelines.
+ - By 1.07.2024: Onboard main stakeholders on the new process.
```
Comment (Rotfuks) [inline technical-decision-making-process/README.md:68]: Just a small form thing but to fit the other date formats I would do "By 01.07.2024", or my OCD gets triggered :D

File: technical-decision-making-process/README.md [hunk 18]
```diff
@@ -0,0 +1,84 @@
...
+- Problem statement
+- Decision maker
+- Deadline
+- Who is affected / stakeholders
+- Preferred solution
+- Alternative solutions (optional)
+- Implementation plan (optional)
+- Communication plan
+
+### Alternative solutions
```
[the remark this answers is not in the document]
Comment (pipo02mix) [reply technical-decision-making-process/README.md:56]: IMO it is just for mentioning which other alternatives have been evaluated (from the surface)

File: technical-decision-making-process/README.md [hunk 19]
```diff
@@ -0,0 +1,84 @@
...
+- Platform Architects
+- Team Horizon
+
+### Preferred solution
+
+#### Technical decision-making process:
+- Proposal Creation: Any team member can create an RFC document to propose a decision. See [RFC and decision making process](https://handbook.giantswarm.io/docs/rfcs/decision-process/) for a description of submitting an RFC.
+- Review and Discussion: The RFC is shared with relevant stakeholders for review and feedback.
+- Decision Meeting: If there is no agreement or insufficient feedback before the deadline, the decision owner schedules a formal meeting with relevant stakeholders to discuss the RFC and make a decision.
+- Approval and Implementation: Once agreed upon, the decision is documented, and an implementation plan is created.
```
Comment (mcharriere) [inline technical-decision-making-process/README.md:43]: As I suggested in the meeting, It'd be nice to include a "feedback/heath check/retrospective" step some time after approval or implementation.

```suggestion
- Approval and Implementation: Once agreed upon, the decision is documented, and an implementation plan is created.
- Feedback: Retrospective sessions with the stakeholders are recommended, during and after the implementation, to evaluate and document the impact of the decision.
```

I don't think we need to go into details, and of course we can improve the process in the future.

File: technical-decision-making-process/README.md (hunk 16 above)
Comment (snizhana-dynnyk) [inline technical-decision-making-process/README.md:18]:
```suggestion
The first [RFC and decision making process](https://handbook.giantswarm.io/docs/rfcs/decision-process/) proposed a file structure and introduced an overview of RFCs in the [Handbook](https://handbook.giantswarm.io/docs/rfcs/).
```

File: technical-decision-making-process/README.md [hunk 20]
```diff
@@ -0,0 +1,85 @@
...
+  - Cons: Inefficiencies and inconsistencies remain.
+
+- Adopt a free-form document process:
+  - Pros: Flexibility in document format.
+  - Cons: Lack of structure can lead to confusion and missed details.
+
+### Implementation Plan
+Preparation Phase:
+ - By 13.06.2024: Develop the RFC template and guidelines.
+ - By 1.07.2024: Onboard main stakeholders on the new process.
```
Comment (snizhana-dynnyk) [inline technical-decision-making-process/README.md:69]:
```suggestion
 - By 01.07.2024: Onboard main stakeholders on the new process.
```
